lib/services/plugins-service.ts

```diff
@@ -36,9 +36,24 @@ export class PluginsService implements IPluginsService {
 	public add(plugin: string): IFuture<void> {
 		return (() => {
 			this.ensure().wait();
-			let dependencyData = this.$npm.cache(plugin, undefined, PluginsService.NPM_CONFIG).wait();
-			if (dependencyData.nativescript) {
-				let pluginData = this.convertToPluginData(dependencyData);
+			let installedModuleNames = this.$npm.install(plugin, this.$projectData.projectDir, PluginsService.NPM_CONFIG);
+
+			//npmInstallResultStream is not very usefull (doesn't contain all the package information), so we need to find the real one inside node_modules
+			let foundValidPlugin = false;
+			let realNpmPackageJson:any;
+			for(let index in installedModuleNames) {
+				let name = installedModuleNames[index];
+				let pathToRealNpmPackageJson = `${this.$projectData.projectDir}/node_modules/${name}/package.json`;
+				realNpmPackageJson = require(pathToRealNpmPackageJson);
```

Comment:
it's safer to use `this.$fs.readJson(pathToRealNpmPackageJson).wait()` as require caches results and also it does not work in case the file has BOM header